pkg/rancher-components/src/components/Form/ToggleSwitch/ToggleSwitch.test.ts

```diff
@@ -0,0 +1,89 @@
+import { shallowMount } from '@vue/test-utils';
+import { ToggleSwitch } from './index';
+
+describe('ToggleSwitch.vue', () => {
+  it('renders falsy by default', () => {
+    const wrapper = shallowMount(ToggleSwitch);
+
+    const toggleInput = wrapper.find('input[type="checkbox"]').element as HTMLInputElement
+
+    expect(toggleInput.checked).toBeFalsy();
+  });
+
+  it('renders truthy', () => {
+    const wrapper = shallowMount(
+      ToggleSwitch,
+      {
+        propsData: {
+          value: true
+        }
+      });
+
+    const toggleInput = wrapper.find('input[type="checkbox"]').element as HTMLInputElement
+
+    expect(toggleInput.checked).toBeTruthy();
+  });
+
+  it('updates from falsy to truthy when props change', async () => {
+    const wrapper = shallowMount(ToggleSwitch);
+  
+    const toggleInput = wrapper.find('input[type="checkbox"]').element as HTMLInputElement
+
+    expect(toggleInput.checked).toBeFalsy();
+
+    await wrapper.setProps({ value: true });
+  
+    expect(toggleInput.checked).toBeTruthy();
+  });
+
+  it('emits an input event with default values when clicked', async () => {
+    const wrapper = shallowMount(ToggleSwitch);
+
+    (wrapper.vm as any).toggle(true);
+
+    await wrapper.vm.$nextTick();
+
+    expect(wrapper.emitted().input).toBeTruthy()
```

Comment:
We should have different tests for different assertions. Also the first assertion may not be required as the second and third may cover it.

Reply:
These tests are simple enough that it's easy to split. 

I think it makes sense to remove the first assertion as well.